tests/integration/components/content-kit-editor/component-test.js

```diff
@@ -215,17 +217,134 @@ test('it de-links selected text and fires `on-change`', function(assert) {
   );
 });
 
-test('it adds a component to the content-kit editor', function(assert) {
-  assert.expect(1);
-  this.registry.register('template:components/demo-card-editor', hbs`DEMO`);
+test('it adds a component in display mode to the content-kit editor', function(assert) {
+  assert.expect(5);
+  this.registry.register('template:components/demo-card', hbs`
+    <div id="demo-card"><button id='edit-card' {{action editCard}}></button></div>
+   `);
+  this.registry.register('template:components/demo-card-editor', hbs`<div id="demo-card-editor"></div>`);
   this.set('cards', [
     createComponentCard('demo-card')
   ]);
   this.render(hbs`
     {{#content-kit-editor cards=cards as |contentKit|}}
-      <button {{action contentKit.addCard 'demo-card'}}></button>
+      <button id='add-card' {{action contentKit.addCard 'demo-card'}}></button>
     {{/content-kit-editor}}
   `);
-  this.$('button').click();
-  assert.ok(!!this.$(`div:contains(DEMO)`).length, 'Card added');
+
+  this.$('button#add-card').click();
+  assert.ok(this.$(`#demo-card`).length, 'Card added in display mode');
+  assert.ok(!this.$(`#demo-card-editor`).length, 'Card not in edit mode');
+  assert.ok(this.$('button#edit-card').length, 'has edit card button');
+
+  this.$('button#edit-card').click();
+  assert.ok(!this.$(`#demo-card`).length, 'Card not in display mode');
+  assert.ok(this.$(`#demo-card-editor`).length, 'Card changed to edit mode');
+});
+
+test('it has `addEditorCard` action to add card in edit mode', function(assert) {
+  assert.expect(2);
+  this.registry.register('template:components/demo-card', hbs`<div id="demo-card"></div>`);
+  this.registry.register('template:components/demo-card-editor', hbs`<div id="demo-card-editor"></div>`);
+  this.set('cards', [
+    createComponentCard('demo-card')
+  ]);
+  this.render(hbs`
+    {{#content-kit-editor cards=cards as |contentKit|}}
+      <button id='add-card' {{action contentKit.addEditorCard 'demo-card'}}></button>
+    {{/content-kit-editor}}
+  `);
+
+  this.$('button#add-card').click();
+
+  assert.ok(!this.$(`#demo-card`).length, 'Card added in edit mode');
+  assert.ok(this.$(`#demo-card-editor`).length, 'Card not in display mode');
+});
+
+test('#activeSectionTagNames is correct', function(assert) {
```

Comment:
the `activeSectionTagNames` tests are not new; they were just in the wrong file before.